src/gui/shell.cpp

```diff
@@ -421,6 +423,11 @@ void Shell::handleInsertMode(QPainter& painter)
 	m_insertMode = true;
 }
 
+void Shell::handleSetTitle(const QVariantList& opargs)
+{
+	emit neovimSetTitle(opargs.at(0).toString());
+}
```

Comment:
Be a bit more pedantic (`QList::at()` needs to be passed a valid index) and take care of the string encoding conversions

```diff
diff --git a/src/gui/shell.cpp b/src/gui/shell.cpp
index 47a34be..aa0d844 100644
--- a/src/gui/shell.cpp
+++ b/src/gui/shell.cpp
@@ -425,7 +425,13 @@ void Shell::handleInsertMode(QPainter& painter)
 
 void Shell::handleSetTitle(const QVariantList& opargs)
 {
-       emit neovimSetTitle(opargs.at(0).toString());
+
+       if (opargs.size() != 1 || !opargs.at(0).canConvert<QByteArray>()) {
+               qWarning() << "Unexpected arguments for set_title:" << opargs;
+               return;
+       }
+       QString title = m_nvim->decode(opargs.at(0).toByteArray());
+       emit neovimSetTitle(title);
 }
```